library/network_connections.py

```diff
@@ -676,7 +676,15 @@ def setting_ip_config_get_routes(self, s_ip):
     def connection_ensure_setting(self, connection, setting_type):
         setting = connection.get_setting(setting_type)
         if not setting:
-            setting = setting_type.new()
+            util_nm = Util.NM()
+            if (
+                hasattr(util_nm, "SettingMatch")
+                and setting_type is util_nm.SettingMatch
+            ):
+                # https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/commit/4d609dd92f2922e92888e02b5b857e4d7b10a37e
```

Comment:
```python
if setting_type is getattr(Util.NM(), 'SettingMatch', None):
    setting = setting_type()
else:
    setting = setting_type.new()
```

Reply:
updated

Reply:
Your idea is to use `g_object_new(NM_TYPE_SETTING_*, NULL)` instead of `nm_setting_*_new()`. Actually, all nm_setting_*_new() function only call g_object_new() directly -- without doing anything else.